src/OrbitGl/App.h

```diff
@@ -579,26 +577,46 @@ class OrbitApp final : public DataViewFactory,
  private:
   void UpdateModulesAbortCaptureIfModuleWithoutBuildIdNeedsReload(
       absl::Span<const orbit_grpc_protos::ModuleInfo> module_infos);
-  void AddSymbols(const orbit_symbol_provider::ModuleIdentifier& module_id,
-                  const orbit_grpc_protos::ModuleSymbols& module_symbols);
   ErrorMessageOr<std::vector<const orbit_client_data::ModuleData*>> GetLoadedModulesByPath(
       const std::filesystem::path& module_path);
   ErrorMessageOr<void> ConvertPresetToNewFormatIfNecessary(
       const orbit_preset_file::PresetFile& preset_file);
 
-  [[nodiscard]] orbit_base::Future<ErrorMessageOr<orbit_base::CanceledOr<std::filesystem::path>>>
-  RetrieveModuleFromRemote(const std::string& module_file_path, orbit_base::StopToken stop_token);
-
   void SelectFunctionsFromHashes(const orbit_client_data::ModuleData* module,
                                  absl::Span<const uint64_t> function_hashes);
   void SelectFunctionsByName(const orbit_client_data::ModuleData* module,
                              absl::Span<const std::string> function_names);
 
   orbit_base::Future<ErrorMessageOr<std::filesystem::path>> FindModuleLocally(
       const orbit_client_data::ModuleData* module_data);
+  // TODO(b/243520787) The following is temporary. The SymbolProvider related logic SHOULD be moved
+  // to the ProxySymbolProvider as planned in our symbol refactoring discussion in b/243520787.
+  void InitRemoteSymbolProviders();
+  // TODO(b/243520787) Rename existing `RetrieveModuleFromRemote` as `RetrieveModuleFromInstance`
+  // and rename this `RetrieveModuleViaDownload` to `RetrieveModuleFromRemote`.
+  [[nodiscard]] orbit_base::Future<ErrorMessageOr<orbit_base::CanceledOr<std::filesystem::path>>>
+  RetrieveModuleViaDownload(const orbit_symbol_provider::ModuleIdentifier& module_id);
+  [[nodiscard]] orbit_base::Future<ErrorMessageOr<orbit_base::CanceledOr<std::filesystem::path>>>
+  RetrieveModuleFromRemote(const std::string& module_file_path, orbit_base::StopToken stop_token);
+  [[nodiscard]] orbit_base::Future<ErrorMessageOr<orbit_base::CanceledOr<std::filesystem::path>>>
```

Comment:
Prefer to put `RetrieveModuleFromRemote` after `FindModuleLocally` such that the order of the symbol sources here is the same as the order we search for symbols.

Reply:
Sorry my mistake, I was treating `RetrieveModuleViaDownload` as symbol stores only 🙈 

Please keep it as is now